tests/Application/Source/StaticReflection/src/MyClass.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+final class MyClass
+{
+    public function __construct()
+    {
+        throw new RuntimeException('Static reflection should not execute the code.');
```

Comment:
@TomasVotruba If we add `list($k, $v) = each($array);` to this method, then it will be refactored during the execution of this test, so it definitely works. I think that a portion of this ticket comes for free just by upgrading PHPStan and putting the relevant config into phpstan.neon: https://github.com/rectorphp/rector/issues/4330